app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt

```diff
@@ -224,48 +229,117 @@ private fun RecentSearchGroupItem(
-    alignment: Alignment = Alignment.Center,
-    icon: Bitmap? = null
+    iconModifier: Modifier = Modifier,
+    icon: Bitmap? = null,
+    thumbnail: Bitmap? = null,
 ) {
-    if (icon != null) {
-        Image(
-            painter = BitmapPainter(icon.asImageBitmap()),
-            contentDescription = null,
-            modifier = modifier,
-            contentScale = contentScale,
-            alignment = alignment
-        )
-    } else {
-        components.core.icons.Loader(
-            url = url
-        ) {
-            Placeholder {
-                Box(
-                    modifier = Modifier.background(
-                        color = when (isSystemInDarkTheme()) {
-                            true -> Color(0xFF42414D) // DarkGrey30
-                            false -> PhotonColors.LightGrey30
-                        }
-                    )
+    Card(
+        modifier = modifier
+    ) {
+        when {
+            thumbnail != null -> {
+                Image(
+                    bitmap = thumbnail.asImageBitmap(),
+                    contentDescription = null,
+                    modifier = modifier,
+                    contentScale = ContentScale.FillWidth,
+                    alignment = Alignment.TopCenter
                 )
             }
-
-            WithIcon { icon ->
+            icon != null -> {
                 Image(
-                    painter = icon.painter,
+                    bitmap = icon.asImageBitmap(),
                     contentDescription = null,
-                    modifier = modifier
+                    modifier = iconModifier,
+                    contentScale = ContentScale.Fit,
+                    alignment = Alignment.Center
                 )
             }
+            else -> {
+                components.core.icons.Loader(
+                    url = url
+                ) {
+                    Placeholder {
+                        Box(
+                            modifier = Modifier.background(
+                                color = when (isSystemInDarkTheme()) {
+                                    true -> Color(0xFF42414D) // DarkGrey30
+                                    false -> PhotonColors.LightGrey30
+                                }
+                            )
+                        )
+                    }
+
+                    WithIcon { icon ->
+                        Image(
+                            painter = icon.painter,
+                            contentDescription = null,
+                            modifier = iconModifier,
+                            contentScale = ContentScale.Fit
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * A recent tab icon.
+ *
+ * @param url The loaded URL of the tab.
+ * @param modifier modifier Modifier used to draw the image content.
+ * @param icon The icon of the tab.
+ * is null.
+ */
+@Composable
+private fun RecentTabIcon(
```

Comment:
Has there been any discussion around doing @Previews for our composeable layouts going forward?

Reply:
There is. I think due to the amount of work recently that is on hold for now.  I agree that we should definitely add @Previews as soon as we can.